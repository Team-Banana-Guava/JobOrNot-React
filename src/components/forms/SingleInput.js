import React from 'react';

const SingleInput = (props) => {
    return(
        <div>
            <label>{props.title}</label>
            <input
                name={props.name}
                type={props.inputType}
                value={props.content}
                onChange={props.controlFunc}
                placeholder={props.placeholder} />
        </div>
    )
}

// SingleInput.propTypes = {
//     title: React.propTypes.string.isRequired,
//     name: React.propTypes.string.isRequired,
//     inputType: React.propTypes.oneOf(['text', 'number']).isRequired,
//     content: React.propTypes.oneOfType([
//         React.propTypes.string,
//         React.propTypes.number,
//     ]).isRequired,
//     placeholder: React.propTypes.string.isRequired,
// }

export default SingleInput;