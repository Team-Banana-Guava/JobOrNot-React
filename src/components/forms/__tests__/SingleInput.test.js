/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SingleInput from '../SingleInput';

describe('SingleInput', function () {
    it('renders the SingleInput field/component', () => {
        const rendered = shallow(<SingleInput title={'string'} name={'string'} inputType={'text'} content={'string'} controlFunc={() => {}} placeholder={'string'}/>);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});
