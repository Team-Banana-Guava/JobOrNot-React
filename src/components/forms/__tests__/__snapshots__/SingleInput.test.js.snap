exports[`SingleInput renders the SingleInput field/component 1`] = `
<div>
  <label>
    string
  </label>
  <input
    name="string"
    onChange={[Function]}
    placeholder="string"
    type="text"
    value="string" />
</div>
`;
